go/libraries/doltcore/sqle/read_replica_database.go

```diff
@@ -215,8 +216,59 @@ func (rrd ReadReplicaDatabase) PullFromRemote(ctx *sql.Context) error {
 	return nil
 }
 
-func (rrd ReadReplicaDatabase) RebaseSourceDb(ctx *sql.Context) error {
-	return rrd.srcDB.Rebase(ctx)
+// CreateLocalBranchFromRemote pulls the given branch from the remote database and creates a local tracking branch for it.
+func (rrd ReadReplicaDatabase) CreateLocalBranchFromRemote(ctx *sql.Context, branchRef ref.BranchRef) error {
+	_, err := rrd.limiter.Run(ctx, "pullNewBranch", func() (any, error) {
```

Comment:
this is maybe subject to a correctness error where two clients try to connect to two different branches, and they will both end up connecting to the branch for whichever calls this func first. I.e. the name needs to be unique to the branch?

Reply:
This is substantially rewritten, plz take another look